src/main/java/seedu/address/model/particpant/Participant.java

```diff
@@ -0,0 +1,213 @@
+package seedu.address.model.particpant;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+
+
+/**
+ * Represents a Participant in an event
+ * Guarantees: name, phone, email, address, tags are present and not null, field values are validated, immutable.
+ * Non-guarantees: birthDate - to be added with builder chaining method
+ */
+public class Participant extends Person {
+
+    private BirthDate birthDate = BirthDate.notSpecified();
+    private Set<Note> notes = new HashSet<>();
+    private ArrayList<Person> nextOfKins = new ArrayList<>();
+
+
+    /**
+     * Every field must be present and not null.
+     *
+     * @param name    Name object of the person
+     * @param phone   Phone object of the person
+     * @param email   Email object of the person
+     * @param address Address object of the person
+     * @param tags    tags of the person
+     */
+    public Participant(Name name, Phone phone, Email email, Address address, Set<Tag> tags) {
+        super(name, phone, email, address, tags);
+    }
+
+    /**
+     * Builder chaining method for retuning participant object with certain birthdate.
+     * This is to allow the use of Person template along with Person but adding extension fields.
+     * i.e. new Participant(name, phone, email, address, tags).withBirthDate(year, month, dayOfMonth)
+     *
+     * @param year       this year or before
+     * @param month      month of the year specified
+     * @param dayOfMonth day of the month specified
+     * @return another Participant with same particulars object but with birth date
+     */
+    public Participant withBirthDate(int year, int month, int dayOfMonth) {
```

Comment:
This method name was quite puzzling to me until I realised that it's meant to attach a birth date to the participant.
Changing the name to addBirthDate will make it more understandable. _(Also it violates the coding standard)_
Same thing can apply to the methods below.

Reply:
yeah, when I think about it it does violate many abstractions do you think I should make a separate class ParticipantBuilder (see **test.java.seedu.address.testutil.PersonBuilder** for reference) so that way we can create a Participant more logically?

I personally don't like having to add different parameters to correct argument position (especially when an object has so many fields to fill the information in) so I really like the idea of using a builder here instead

Reply:
But that class is to build a Person with preallocated arguments. In our case we definitely have to pass a lot of arguments to initialise a Participant, so that one is unavoidable. I think it's fine honestly to have methods to attach more fields. I also think that a constructor that initialises **all** the fields would be useful as well, along with the current one that only initialises for Person superclass.

Reply:
yeap I think I am going to go with that approach instead also I just added a comment in the PR once you are free, feel free to read it and give your opinion :)